pkg/apis/imagepolicy/types.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package imagepolicy
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/api/v1"
+)
+
+// +genclient=true
+// +nonNamespaced=true
+// +noMethods=true
+
+// ImageReview checks if the set of images in a pod are allowed.
+type ImageReview struct {
+	unversioned.TypeMeta
+	v1.ObjectMeta
+
+	// Spec holds information about the pod being evaluated
+	Spec ImageReviewSpec
+
+	// Status is filled in by the backend and indicates whether the pod should be allowed.
+	Status ImageReviewStatus
+}
+
+// ImageReviewSpec is a description of the pod creation request.
+type ImageReviewSpec struct {
+	// Container is a subset of the information in theh container being created.
+	Container ImageReviewContainerSpec
```

Comment:
I think that cache-ability is the strongest argument here, however I'd bet on the vast majority of the caching benefit being at the pod level since storms will probably be RCs, Ds, Jobs, etc scaling up and stamping out the same pod over and over.  

If this were a local call, I'd definitely agree with the change you've made.  As a remote call that's going to be in front of every incoming pod in the system, I'd really prefer to avoid a fanout of remote calls.